core/templates/dev/head/services/contextual/UrlServiceSpec.js

```diff
@@ -88,7 +88,7 @@ describe('Url Service', function() {
   });
 
   it('should correctly retrieve topic id from url', function() {
-    mockLocation.pathname = '/topic_editor/abcdefgijklm#';
+    mockLocation.pathname = '/topic_editor/abcdefgijklm';
```

Comment:
Also add tests for pathnames with hashes, invalid pathnames, etc.

Reply:
I'll do the invalid pathnames, but wouldn't the browser automatically separate the hashes into a window.location.hash field? Since, we are mocking window.location itself, I don't think adding hashes can be tested effectively, as we can just give the hash value to the window.location.hash field.

Reply:
Oh, I see, sorry. Yeah, you may be right. I'm not sure; you'll need to check.

Might be better to mock the whole location instead of just its pathname though.

Reply:
I added a test for invalid pathnames. 
But even with mocking a whole location, the getTopicIdFromUrl() function calls the already existing getPathname() function, so in the end, it would only check the pathname field of window.location.